### ISSUE TITLE
dsolve fails on particular intergral, ignores some bits of forcing

### ISSUE THREAD

**Issue by cbm755** [cbm755/octsympy#183]
Define an inhomogenous linear system of ODEs:

```
octave:52> syms x(t) y(t)
octave:53> e1 = diff(x,t) == x+4*y - t + 2
e1 = (sym)

  d                                
  ──(x(t)) = -t + x(t) + 4⋅y(t) + 2
  dt                               

octave:54> e2 = diff(y,t) == x - 2*y + 5*t
e2 = (sym)

  d                             
  ──(y(t)) = 5⋅t + x(t) - 2⋅y(t)
  dt                            

octave:55> S = dsolve([e1, e2])
S = 
{
 = (sym)

               -3⋅t         2⋅t   2
  x(t) = 4⋅C₁⋅ℯ     + 4⋅C₂⋅ℯ    - ─
                                  3

 = (sym)

                 -3⋅t       2⋅t   1
  y(t) = - 4⋅C₁⋅ℯ     + C₂⋅ℯ    - ─
                                  3

}
octave:70> X = rhs(S{1}),  Y = rhs(S{2})   # extract solutions
X = (sym)

        -3⋅t         2⋅t   2
  4⋅C₁⋅ℯ     + 4⋅C₂⋅ℯ    - ─
                           3

Y = (sym)

          -3⋅t       2⋅t   1
  - 4⋅C₁⋅ℯ     + C₂⋅ℯ    - ─
                           3

octave:71> 
octave:71> diff(X,t) - (X+4*Y - t + 2)   # should be zero
ans = (sym) t
octave:72> diff(Y,t) - (X - 2*Y + 5*t)
ans = (sym) -5⋅t
```

Notice its as if the `t` terms did not appear in the forcing.

Haven't yet checked if it happens for scalars or just systems.

@maprieto any thoughts?

**Comment by cbm755** [comment]
seems ok for scalar:
````
octave:81> e1 = diff(x,t) == x + 2*t + 2
e1 = (sym)

  d                        
  ──(x(t)) = 2⋅t + x(t) + 2
  dt                       

octave:82> dsolve(e1)
ans = (sym)

         ⎛                 -t⎞  t
  x(t) = ⎝C₁ + 2⋅(-t - 2)⋅ℯ  ⎠⋅ℯ 

octave:83> X = rhs(ans)
X = (sym)

  ⎛                 -t⎞  t
  ⎝C₁ + 2⋅(-t - 2)⋅ℯ  ⎠⋅ℯ 

octave:84> diff(X) - (X + 2*t + 2)
ans = (sym)

         ⎛              -t      -t⎞  t    
  -2⋅t + ⎝- 2⋅(-t - 2)⋅ℯ   - 2⋅ℯ  ⎠⋅ℯ  - 2

octave:85> simplify(ans)
ans = (sym) 0
````

**Comment by cbm755** [comment]
Simpler example, diagonal system, there the `2*sin(t)` is ignored but not the `2`.
````
octave:110> e1 = diff(x,t) == x + 2*sin(t) + 2
e1 = (sym)

  d                             
  ──(x(t)) = x(t) + 2⋅sin(t) + 2
  dt                            

octave:111> e2 = diff(y,t) == y
e2 = (sym)

  d              
  ──(y(t)) = y(t)
  dt             

octave:112> S = dsolve([e1, e2])
S = 
{
 = (sym)

             t    
  x(t) = C₂⋅ℯ  - 2

 = (sym)

             t
  y(t) = C₁⋅ℯ 

}
octave:113> 
````

**Comment by maprieto** [comment]
Even a simpler example cannot be resolved from sympy 0.7.6:
```python
import sympy as sp
from sympy.abc import t
x=sp.Function('x')
y=sp.Function('y') 
odes = [sp.Eq(sp.diff(x(t),t), 3*t), sp.Eq(sp.diff(y(t),t),y(t))]
sol = sp.dsolve(odes)
X=sol[0].rhs
Y=sol[1].rhs
sp.diff(X,t) - (3*t) 
sp.diff(Y,t) - Y
```
So, I can confirm this issue and propose to be labelled as upstream. Anyway, I'll try some kind of workaround for system of odes (at least for linear systems with constant coefficients)

**Comment by cbm755** [comment]
Thanks @maprieto I reported upstream as
https://github.com/sympy/sympy/issues/8859

I wouldn't try too hard to workaround it here (unless easy and you need for your course).  Probably better to get it fixed upstream if possible.
